src/lambda_handler.py

```diff
@@ -109,19 +253,79 @@ def handler(event, context) -> dict[str, Optional[str]]:
         old_log_level = logging.getLogger().getEffectiveLevel()
         logging.getLogger().setLevel(new_log_level)
 
-    task_name = f"task_{event.get('task')}"
-    task = globals().get(task_name, task_default)
+    # Set up the SSM client if necessary
+    if ssm_client is None:
+        ssm_client = boto3_client("ssm")
+
+    mongodb_uri_elements: List[Tuple[str, Optional[str]]] = []
+
+    # This only runs from a CloudWatch scheduled event invocation
+    trigger_source: Optional[str]
+    trigger_type: Optional[str]
+    if (
+        (trigger_source := event.get("source")) is None
+        or trigger_source != "aws.events"
+    ) or (
+        (trigger_type := event.get("detail-type")) is None
+        or trigger_type != "Scheduled Event"
+    ):
+        logging.error("Invalid invocation event.")
+        return
+
+    for var in [
+        "ssm_db_user",
+        "ssm_db_pass",
+        "ssm_db_host",
+        "ssm_db_port",
```

Comment:
Moving these away from SSM since they are not particularly sensitive and wouldn't otherwise be stored in SSM Parameter Store. I'm pretty sure this was sourced from code I wrote so apologies for the change request, but when implementing the Terraform infrastructure to deploy this it felt needlessly convoluted.

```suggestion
        "db_host",
        "db_port",
```